### PR TITLE
GSYE-542: Replace myco term occurrences to matching engine.

### PR DESCRIPTION
**Pull request by FeliceRocchitelli**
## Reason for the proposed changes

Replace myco term occurrences to matching engine.

## Proposed changes

-

**INTEGRATION_TESTS_BRANCH**=feature/GSYE-542
**GSY_FRAMEWORK_BRANCH**=master

### CODE REVIEW

**Comment by spyrostz** [comment]
@FeliceRocchitelli just a heads up, in order for CI to pass you need to also fix the occurrences in gsy-framework and gsy-backend-integration-tests repos in order to also not include the term Myco. Once fixed in separate branches / PRs, you can then adapt these variables in this PR description:
```
INTEGRATION_TESTS_BRANCH=master
GSY_FRAMEWORK_BRANCH=master
```
in order to point to the fixed branches. That way the CI will detect the correct branches and will use them correctly. 
On another note I cc'ed @gridsingularity/backend-reviewers  in the PR in order for the rest of the team to have a look too!

**Comment by FeliceRocchitelli** [comment]
> @FeliceRocchitelli just a heads up, in order for CI to pass you need to also fix the occurrences in gsy-framework and gsy-backend-integration-tests repos in order to also not include the term Myco. Once fixed in separate branches / PRs, you can then adapt these variables in this PR description:
> 
> ```
> INTEGRATION_TESTS_BRANCH=master
> GSY_FRAMEWORK_BRANCH=master
> ```
> 
> in order to point to the fixed branches. That way the CI will detect the correct branches and will use them correctly. On another note I cc'ed @gridsingularity/backend-reviewers in the PR in order for the rest of the team to have a look too!

@spyrostz the gsy-framework doesn't have any occurrences. 
I'll take care of changing this gsy-backend-integration-tests and I'll update this PR

File: src/gsy_e/models/matching_engine_matcher/matching_engine_external_matcher.py [hunk 1]
```diff
@@ -26,11 +26,14 @@
 from gsy_framework.enums import AvailableMarketTypes
 
 import gsy_e.constants
-from gsy_e.gsy_e_core.exceptions import InvalidBidOfferPairException, MycoValidationException
+from gsy_e.gsy_e_core.exceptions import (InvalidBidOfferPairException,
+                                         MatchingEngineValidationException)
 from gsy_e.gsy_e_core.market_counters import ExternalTickCounter
-from gsy_e.gsy_e_core.redis_connections.area_market import myco_redis_communicator_factory
+from gsy_e.gsy_e_core.redis_connections.area_market import \
```
Comment (hannesdiedrich) [inline src/gsy_e/models/matching_engine_matcher/matching_engine_external_matcher.py:32]: Please do not use backslashes for lines that exceed the maximum character limit. Please follow the approach that you did in line 29, use brackets. 
Thanks a lot!

File: src/gsy_e/models/myco_matcher/__init__.py [hunk 2]
```diff
@@ -1,9 +0,0 @@
-from gsy_e.models.myco_matcher.myco_internal_matcher import MycoInternalMatcher
```
Comment (hannesdiedrich) [inline src/gsy_e/models/myco_matcher/__init__.py:1]: Can you please explain why you removed this module?
It helps to import the MatchingEngines simpler with
```
from gsy_e.models.myco_matcher import MatchingEngineExternalMatcher
```
Please also add the `MatchingEngineInternalForwardMarketMatcher` 
Thanks!

File: src/gsy_e/models/matching_engine_matcher/__init__.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    MatchingEngineInternalMatcher
+from gsy_e.models.matching_engine_matcher.matching_engine_external_matcher import \
+    MatchingEngineExternalMatcher
+from gsy_e.models.matching_engine_matcher.matching_engine_matcher_interface import \
+    MatchingEngineMatcherInterface
+
+__all__ = [
+    "MatchingEngineMatcherInterface",
+    "MatchingEngineExternalMatcher",
+    "MatchingEngineInternalMatcher"
```
Comment (hannesdiedrich) [inline src/gsy_e/models/matching_engine_matcher/__init__.py:11]: Please also add the `MatchingEngineInternalForwardMarketMatcher`. 
Thanks a lot!

**Review by hannesdiedrich — COMMENTED** [review]
Two minor comments, LGTM once adapted

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**